pkg/agent/run.go

```diff
@@ -183,11 +183,14 @@ func Run(cmd *cobra.Command, args []string) error {
 
 		// start the data gatherers and wait for the cache sync
 		group.Go(func() error {
+			// Most implementations of `DataGatherer.Run` return immediately.
```

Comment:
Side-note, you don't need to change anything:

I find it weird that the `Run` func's implementation varies from data gatherers to data gatherers... 

The specification of the `Run` func (an interface, I assume) seems to be defined as "Run blocks until the channel is closed".

But the local datagatherer, for example, is implemented like this:

```go
func (g *DataGatherer) Run(stopCh <-chan struct{}) error {
	// no async functionality, see Fetch
	return nil
}
```

It should be:

```go
func (g *DataGatherer) Run(stopCh <-chan struct{}) error {
	// no async functionality, see Fetch
	<-stopCh
	return nil
}
```